File: backend-python/app/modules/system/controller/auth_controller.py
from fastapi import APIRouter, Depends, Request

from app.common.api_response import ApiResponse, success
from app.modules.system.dto.auth_request import LoginRequest, PasswordRequest, ProfileUpdateRequest
from app.modules.system.router import current_session, get_auth_service
from app.modules.system.service.auth_service import AuthService

router = APIRouter(tags=["auth"])


@router.post("/login", response_model=ApiResponse)
async def login(request: Request, body: LoginRequest, service: AuthService = Depends(get_auth_service)) -> ApiResponse:
    return success(await service.login(body, request.client.host if request.client else "", request.headers.get("User-Agent", "")))


@router.post("/logout", response_model=ApiResponse)
async def logout(session: dict = Depends(current_session), service: AuthService = Depends(get_auth_service)) -> ApiResponse:
    await service.logout(session)
    return success()


@router.get("/me", response_model=ApiResponse)
async def me(session: dict = Depends(current_session), service: AuthService = Depends(get_auth_service)) -> ApiResponse:
    return success(await service.me(session))


@router.put("/password", response_model=ApiResponse)
async def change_password(
    body: PasswordRequest, session: dict = Depends(current_session), service: AuthService = Depends(get_auth_service)
) -> ApiResponse:
    await service.change_password(session, body)
    return success()


@router.put("/profile", response_model=ApiResponse)
async def update_profile(
    body: ProfileUpdateRequest, session: dict = Depends(current_session), service: AuthService = Depends(get_auth_service)
) -> ApiResponse:
    await service.update_profile(session, body)
    return success()

File: backend-rust/src/modules/system/controller/auth_controller.rs
use super::{ok, ok_null};
use crate::common::{ApiResponse, AppError};
use crate::modules::system::AppState;
use crate::modules::system::dto::auth_request::{
    LoginRequest, PasswordRequest, ProfileUpdateRequest,
};
use crate::modules::system::service::auth_service;
use crate::modules::system::vo::auth_vo::{AuthLoginVo, AuthMeVo};
use axum::Json;
use axum::extract::State;
use axum::extract::rejection::JsonRejection;
use axum::http::HeaderMap;

pub async fn login(
    State(state): State<AppState>,
    payload: Result<Json<LoginRequest>, JsonRejection>,
) -> Result<Json<ApiResponse<AuthLoginVo>>, AppError> {
    let Json(request) = payload.map_err(|_| AppError::bad_request("请求体 JSON 格式错误"))?;
    let _ = &state.settings.token;
    Ok(ok(auth_service::login(&state, request, &HeaderMap::new()).await?))
}

pub async fn logout(
    State(state): State<AppState>,
    headers: HeaderMap,
) -> Result<Json<ApiResponse<()>>, AppError> {
    auth_service::logout(&state, &headers).await?;
    Ok(ok_null())
}

pub async fn me(
    State(state): State<AppState>,
    headers: HeaderMap,
) -> Result<Json<ApiResponse<AuthMeVo>>, AppError> {
    Ok(ok(auth_service::me(&state, &headers).await?))
}

pub async fn password(
    State(state): State<AppState>,
    headers: HeaderMap,
    Json(request): Json<PasswordRequest>,
) -> Result<Json<ApiResponse<()>>, AppError> {
    auth_service::change_password(&state, &headers, request).await?;
    Ok(ok_null())
}

pub async fn profile(
    State(state): State<AppState>,
    headers: HeaderMap,
    Json(request): Json<ProfileUpdateRequest>,
) -> Result<Json<ApiResponse<()>>, AppError> {
    auth_service::update_profile(&state, &headers, request).await?;
    Ok(ok_null())
}
